### PR TITLE
[azure] add functions notebook sample

### PR DESCRIPTION
**Pull request by kristapratico**

### CODE REVIEW

File: examples/azure/functions.ipynb [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if not use_azure_active_directory:\n",
+    "    openai.api_type = \"azure\"\n",
+    "    openai.api_key = \"\"  # Add your api key here"
```
Comment (ted-at-openai) [inline examples/azure/functions.ipynb:109]: Happy to merge.

One thing I noticed here is that we're encouraging developers to put their API keys in code. In general, I really want to discourage this, as the leak of an API key can be quite costly. Do you mind removing this and using environment variables instead?

**Review by ted-at-openai — APPROVED** [review]

File: examples/azure/functions.ipynb (hunk 1 above)
Comment (kristapratico) [reply examples/azure/functions.ipynb]: Thanks for reviewing, that's good feedback. I updated the code to use an environment variable for API key across all the azure examples.

Comment (ted-at-openai) [reply examples/azure/functions.ipynb]: Awesome. Thank you!

**Comment by ted-at-openai** [comment]
Merged!

**Pull request merged**